### PR TITLE
feat: start reading the Reference Include property

### PR DESCRIPTION
**Pull request by lili2311**
- Start saving more deps that are definedin `*.csproj` files in format:

```
<Reference Include="Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
      <HintPath>..\packages\Newtonsoft.Json.10.0.3\lib\net45\Newtonsoft.Json.dll</HintPath
</Reference>

```
and 
```   
 <Reference Include="System.Web.ApplicationServices" />
```

- Added some tests

Note: dependency Type is for now assumed prod until we determine correct way to tell it is a dev dep

### CODE REVIEW

File: lib/parsers/index.ts [hunk 1]
```diff
@@ -1,10 +1,5 @@
 import * as parseXML from 'xml2js';
 import * as _ from 'lodash';
-export interface Dep {
-  name: string;
-  version: string;
-  dev?: boolean;
-}
```
Comment (lili2311) [inline lib/parsers/index.ts]: this seems unused

File: lib/parsers/index.ts [hunk 2]
```diff
@@ -22,6 +17,13 @@ export enum DepType {
   dev = 'dev',
 }
 
+export interface ReferenceInclude {
```
Comment (lili2311) [inline lib/parsers/index.ts]: note the name of package doesn't have a key! it is the first string until `,`

File: lib/parsers/index.ts [hunk 3]
```diff
@@ -22,6 +17,18 @@ export enum DepType {
...
+  Version?: string;
+  Culture?: string;
+  processorArchitecture?: string;
+  PublicKeyToken?: string;
+}
+
+export interface DependenciesDiscoveryResult {
```
Comment (miiila) [inline lib/parsers/index.ts]: Do we need to export this interface? In addition, do we need to export `getDependenciesFromReferenceInclude` and `getDependenciesFromPackageReference`?

Comment (lili2311) [reply lib/parsers/index.ts]: we don't i though i might want to test it separately

**Pull request merged**